odl/contrib/fom/supervised.py

```diff
@@ -201,22 +215,32 @@ def mean_value_difference(data, ground_truth, mask=None, normalized=False,
 
     The normalized variant takes values in :math:`[0, 1]`.
     """
-    l1_norm = odl.solvers.L1Norm(data.space)
+    try:
```

Comment:
Isn't this basically `odl.vector` with a check?

Reply:
Could very well be.

Reply:
An option is to change `odl.vector` to not cast anything if it is already a vector, then just call it here.

Reply:
Perhaps add a `odl.asvector` to be exactly consistent with numpy

Reply:
Nah, it's a crappy function anyway. I don't want to add more crappiness.

Reply:
I've updated the code, should be better now.